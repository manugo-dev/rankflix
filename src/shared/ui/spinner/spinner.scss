.spinner {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  background: #0a0a14;
}

.eye {
  position: relative;
  z-index: 1;
  overflow: hidden;
  width: 78px;
  height: 78px;
  box-sizing: border-box;
  border: 8px solid #131a1d;
  border-radius: 50%;
  margin: 10rem auto;
  background: #fff;
}

.eye::before {
  position: absolute;
  z-index: 2;
  bottom: 15px;
  left: 20px;
  width: 32px;
  height: 32px;
  border-radius: 50%;
  animation: eye-move 3s infinite;
  background: #111;
  content: "";
}

.eye::after {
  position: absolute;
  z-index: 5;
  top: -50%;
  left: 0;
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  border-bottom: 8px solid #131a1d;
  animation: eye-shade 3s infinite;
  background: #263238;
  content: "";
}

@keyframes halo-pulse {
  0%,
  100% {
    opacity: 0.3;
    transform: scale(0.2);
  }

  50% {
    opacity: 0.5;
    transform: scale(1);
  }
}

/* animaciones del ojo */
@keyframes eye-shade {
  0% {
    transform: translateY(0);
  }

  20% {
    transform: translateY(5px);
  }

  40%,
  50% {
    transform: translateY(-5px);
  }

  60% {
    transform: translateY(-8px);
  }

  75% {
    transform: translateY(5px);
  }

  100% {
    transform: translateY(10px);
  }
}

@keyframes eye-move {
  0% {
    transform: translate(0, 0);
  }

  20% {
    transform: translate(0, 5px);
  }

  40%,
  50% {
    transform: translate(0, -5px);
  }

  60% {
    transform: translate(-10px, -5px);
  }

  75% {
    transform: translate(-20px, 5px);
  }

  100% {
    transform: translate(0, 10px);
  }
}
