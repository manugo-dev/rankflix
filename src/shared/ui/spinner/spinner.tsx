import "./spinner.scss";

export function Spinner() {
  return (
    <div className="spinner" role="status">
      <div className="eye"></div>
      <div className="halo"></div>
    </div>
  );
}
