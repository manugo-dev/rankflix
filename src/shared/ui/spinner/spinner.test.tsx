import { cleanup, render, screen } from "@testing-library/react";
import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";

import { Spinner } from "./spinner";

const DEFAULT_PROPS = {};

function renderSpinner() {
  return render(<Spinner {...DEFAULT_PROPS} />);
}

describe("Spinner", () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  afterEach(() => {
    cleanup();
  });

  it("matches the snapshot", () => {
    const { container } = renderSpinner();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the expected DOM structure", () => {
    renderSpinner();

    const outer = screen.queryByRole("status") ?? document.querySelector(".spinner");

    expect(outer).toBeTruthy();
    expect(document.querySelector(".spinner .eye")).toBeTruthy();
    expect(document.querySelector(".spinner .halo")).toBeTruthy();
  });
});
