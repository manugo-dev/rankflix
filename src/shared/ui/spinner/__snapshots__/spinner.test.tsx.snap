// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Spinner > matches the snapshot 1`] = `
<div
  class="spinner"
  role="status"
>
  <div
    class="eye"
  />
  <div
    class="halo"
  />
</div>
`;
